src/util/rotary.cpp

```diff
@@ -5,19 +5,18 @@
 constexpr int kiRotaryFilterMaxLen = 50;
 
 Rotary::Rotary()
-    : m_iFilterPos(0),
-      m_dCalibration(1.0),
-      m_dLastValue(0.0),
-      m_iCalibrationCount(0) {
-    m_iFilterLength = kiRotaryFilterMaxLen;
-    m_pFilter = new double[m_iFilterLength];
+        : m_iFilterLength(kiRotaryFilterMaxLen),
+          m_iFilterPos(0),
+          m_pFilter(std::vector<double>(m_iFilterLength)),
+          m_dCalibration(1.0),
+          m_dLastValue(0.0),
+          m_iCalibrationCount(0) {
     for (int i = 0; i < m_iFilterLength; ++i) {
         m_pFilter[i] = 0.;
     }
```

Comment:
```suggestion
          m_pFilter(std::vector<double>(m_iFilterLength, 0.0)),
          m_dCalibration(1.0),
          m_dLastValue(0.0),
          m_iCalibrationCount(0) {
```
we can get rid of the manual initialization loop if we rely on the constructor to do this.

Reply:
`m_pFilter{m_iFilterLength, 0.0}`

Reply:
["uniform" != uniform in c++ world of fun...](https://github.com/mixxxdj/mixxx/pull/4341/files#r720110255)